rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -155,12 +156,19 @@ void SequentialCompressionWriter::compress_last_file()
 
 void SequentialCompressionWriter::split_bagfile()
 {
+  // Flush buffer layer
+  buffer_layer_->close();
+
   storage_options_.uri = format_storage_uri(
     base_folder_,
     metadata_.relative_file_paths.size());
 
   storage_ = storage_factory_->open_read_write(storage_options_);
 
+  // Update storage in buffer layer and restart consumer thread
+  buffer_layer_->set_storage(storage_);
```

Comment:
I am somehow having a problem with this chain of dependency. I don't have a real argument for this, but I would feel better about it if the storage would not belong to the buffer.
Can we have the consumer/producer threads operate on the buffer, but not belong to them?